SpeechResponder/Service/ScriptRecoveryService.cs

```diff
@@ -0,0 +1,118 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+
+namespace EddiSpeechResponder.Service
+{
+	public class ScriptRecoveryService
+	{
+		static ScriptRecoveryService()
+		{
+			WorkingDirectory = Directory.GetCurrentDirectory();
+		}
+
+		public ScriptRecoveryService(EditScriptWindow scriptWindow)
+		{
+			_scriptWindow = scriptWindow;
+			_lockRoot = new object();
+		}
+
+		private static readonly string WorkingDirectory;
+		private string _tempFileName;
+		private EditScriptWindow _scriptWindow;
+		private bool _scriptSaveCallGuard;
+		private readonly object _lockRoot;
+
+		public static Script GetRecoveredScript()
+		{
+			var recoveringScript = Directory.EnumerateFiles(WorkingDirectory, "*.temp").FirstOrDefault();
+			if (recoveringScript == null)
+			{
+				return null;
+			}
+
+			return JsonConvert.DeserializeObject<Script>(File.ReadAllText(recoveringScript));
+		}
+
+		/// <summary>
+		///		Will be called when ether the name of the script has changed or the script edit window was opened
+		/// </summary>
+		/// <param name="scriptName"></param>
+		public void BeginScriptRecovery(string scriptName)
+		{
+			_tempFileName = Path.Combine(WorkingDirectory, scriptName + ".temp");
```

Comment:
This can result in an exception if the script name includes illegal characters in the path. For example, if I name a script `<test>` then try to edit it, the ScriptRecoveryService throws an unhandled exception and EDDI crashes.

Reply:
I have switched to a constant backup file name